glslc/src/file_compiler.cc

```diff
@@ -170,30 +171,42 @@ bool FileCompiler::EmitCompiledResult(
     }
   }
 
-  // Should write the result as a C-style initializer list if:
-  //  1) the result is SPIRV binary code;
-  //  2) the compiler is set to emit result as a C-style initializer list;
-  //  3) the compiler is not dumping dependency info as compilation output.
-  const bool should_emit_spirv_binary_as_c_initializer_list =
-      std::is_same<CompilationResultType,
-                   shaderc::SpvCompilationResult>::value &&
-      emit_binary_as_c_init_list_ &&
-      !(dependency_info_dumping_handler_ &&
-        dependency_info_dumping_handler_->DumpingAsCompilationOutput());
-
-  // Write compilation output to output file.
-  if (should_emit_spirv_binary_as_c_initializer_list) {
-    // The output should be written to the output stream as a C-style
-    // initializer list.
-    *out << "{";
-    for (uint32_t word : result) {
-      *out << "0x" << std::uppercase << std::setfill('0') << std::setw(8)
-           << std::hex << word << ",";
-    }
-    *out << "}" << std::endl;
-  } else {
-    out->write(compilation_output.data(), compilation_output.size());
+  // Write compilation output to output file. If an output format for SPIR-V
+  // binary code is specified, it is handled here.
+  switch(binary_emission_format_) {
+    case SpirvBinaryEmissionFormat::Unspecified:
+    case SpirvBinaryEmissionFormat::Binary:
+      // The output format is unspecified or specified as binary output.
+      out->write(compilation_output.data(), compilation_output.size());
+      break;
+    case SpirvBinaryEmissionFormat::Numbers:
+    case SpirvBinaryEmissionFormat::CInitList:
+      // The output format is specified to be a list of numbers or a C-style
+      // initializer list. In such cases the compilation output must be in
+      // SPIR-V binary code form.
+      assert(output_type_ == OutputType::SpirvBinary);
+      if (binary_emission_format_ == SpirvBinaryEmissionFormat::CInitList) {
+        *out << "{";
+      }
+      std::ios::fmtflags output_stream_flag_cache(out->flags());
+      *out << std::hex << std::setfill('0');
+      auto RI = result.begin();
+      *out << "0x" << std::setw(8) << *RI++;
```

Comment:
This does trigger a seg fault when the compilation fails...
Fixed in d7942c13f40e2efafe1ea526b929cc074f6a6280